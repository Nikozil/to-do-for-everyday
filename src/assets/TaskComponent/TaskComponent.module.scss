@import '../../App.scss';

.content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  color: inherit;

  @media screen and (max-width: 767px) {
    padding-right: 10px;
  }
}
.content_border {
  box-shadow: 1px 1px 5px $secondary-color;
  // background-color: rgba(29, 52, 156, 0.205);
  border-radius: 5px;
  padding: 2px 6px;
}
