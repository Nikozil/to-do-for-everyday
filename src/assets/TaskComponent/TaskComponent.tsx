import React, { ReactNode } from 'react';
import styles from './TaskComponent.module.scss';

const TaskComponent: React.FC<PropsType> = ({ children }) => {
  return <li className={styles.content}>{children}</li>;
};

export default TaskComponent;

interface PropsType {
  children: ReactNode;
}
