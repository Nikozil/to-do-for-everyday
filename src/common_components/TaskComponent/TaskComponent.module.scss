@import '../../App.scss';

.content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  color: inherit;
  text-align: left;
  z-index: 2000;
  background-color: $content-color;

  @media screen and (max-width: 767px) {
    padding-right: 15px !important;
    padding-left: 10px !important;
  }
}
.content_border {
  box-shadow: 1px 1px 5px $secondary-color;
  border-radius: 5px;
  padding: 2px 6px;
}
