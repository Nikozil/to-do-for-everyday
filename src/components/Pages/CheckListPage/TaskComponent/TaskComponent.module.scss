@import '../../../../App.scss';

.content {
  display: flex;
  justify-content: space-between;
  // align-items: center;
  box-sizing: border-box;

  // font-size: 1.5em;
  color: $primary-color;
  // @media screen and (max-width: 767px) {
  //   font-size: 1rem;
  // }
}
.doneToggle {
  cursor: pointer;
}
