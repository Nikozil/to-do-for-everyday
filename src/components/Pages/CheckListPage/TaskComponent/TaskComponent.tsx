import React from 'react';
import styles from './TaskComponent.module.scss';
import {
  ImCheckboxUnchecked,
  ImCheckboxChecked,
  ImCross,
} from 'react-icons/im';
import { PartialTaskData } from '../../../../Redux/modules/tasksSlice';

const TaskComponent: React.FC<PropsType> = ({
  id,
  name,
  done,
  checkHandler,
  deleteHandler,
}) => {
  const clickCheckHandler = (event: React.MouseEvent) => {
    event.preventDefault();
    checkHandler(id, { done: !done });
  };
  const clickDeleteHandler = (event: React.MouseEvent) => {
    event.preventDefault();
    deleteHandler(id);
  };
  return (
    <div className={styles.content}>
      <span className={styles.task}>
        <button
          name={'delete task'}
          onClick={clickDeleteHandler}
          className={styles.doneToggle}
          data-testid="delete-button">
          <ImCross />
        </button>
        <span className={styles.taskName}>{name}</span>
      </span>
      <button
        name={'task done'}
        onClick={clickCheckHandler}
        className={styles.doneToggle}
        data-testid="check-button">
        {done ? <ImCheckboxChecked /> : <ImCheckboxUnchecked />}
      </button>
    </div>
  );
};

export default TaskComponent;

interface PropsType {
  id: string;
  name: string;
  done: boolean;
  checkHandler: (id: string, data: PartialTaskData) => void;
  deleteHandler: (id: string) => void;
}
