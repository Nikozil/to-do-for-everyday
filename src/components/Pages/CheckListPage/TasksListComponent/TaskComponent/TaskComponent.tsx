import cn from 'classnames';
import React from 'react';
import { ImCheckboxUnchecked, ImCross } from 'react-icons/im';
import { RiPushpinFill, RiPushpinLine } from 'react-icons/ri';
import { PartialTaskData, Task } from '../../../../../Redux/modules/tasksSlice';
import styles from './TaskComponent.module.scss';

const TaskComponent: React.FC<PropsType> = ({
  task,
  checkHandler,
  repeatHandler,
  deleteHandler,
}) => {
  const { id, data } = task;
  const { name, repeat } = data;

  const clickCheckHandler = (event: React.MouseEvent) => {
    event.preventDefault();
    checkHandler(task);
  };
  const clickRepeatHandler = (event: React.MouseEvent) => {
    event.preventDefault();

    repeatHandler(id, { repeat: repeat ? 0 : 1 });
  };
  const clickDeleteHandler = (event: React.MouseEvent) => {
    event.preventDefault();
    deleteHandler(id);
  };

  return (
    <li className={styles.content}>
      <span className={styles.task}>
        <button
          aria-label={'delete task'}
          name={'delete task'}
          onClick={clickDeleteHandler}
          className={styles.button}
          data-testid="delete-button">
          <ImCross />
        </button>
        <span className={styles.taskName}>{name}</span>
      </span>

      <span className={styles.buttons}>
        <button
          aria-label={'repeat task'}
          name={'repeat task'}
          onClick={clickRepeatHandler}
          className={styles.button}
          data-testid="repeat-button">
          {repeat ? <RiPushpinFill /> : <RiPushpinLine />}
        </button>
        <button
          aria-label={'task check'}
          name={'task check'}
          onClick={clickCheckHandler}
          className={cn(styles.button, styles['buttons-margin'])}
          data-testid="check-button">
          <ImCheckboxUnchecked />
        </button>
      </span>
    </li>
  );
};

export default TaskComponent;

interface PropsType {
  task: Task;
  checkHandler: (task: Task) => void;
  repeatHandler: (id: string, data: PartialTaskData) => void;
  deleteHandler: (id: string) => void;
}
