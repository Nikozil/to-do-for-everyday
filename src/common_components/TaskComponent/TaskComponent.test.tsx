import { render } from '../../utils/tests/test-utils';
import TaskComponent from './TaskComponent';

describe('Task Component tests', () => {
  it('rendering and click delete', async () => {
    const { asFragment } = render(<TaskComponent>Hello</TaskComponent>);
    expect(asFragment()).toMatchSnapshot();
  });
});
