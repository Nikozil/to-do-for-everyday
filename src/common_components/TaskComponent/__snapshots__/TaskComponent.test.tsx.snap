// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Component tests rendering and click delete 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    Hello
  </div>
   
</DocumentFragment>
`;
